Core/Core/UITestHelpers/UITestHelpers.swift

```diff
@@ -130,9 +130,13 @@ public class UITestHelpers {
         }
     }
 
-    @discardableResult
-    func send(_ message: IPCDriverServerMessage) -> Data? {
-        return try? ipcDriverClient?.requestRemote(message)
+    private let ipcQueue = DispatchQueue(label: "UITestHelper-ipc")
+    private let callbackQueue = DispatchQueue(label: "UITestHelper-callback")
+    func send(_ message: IPCDriverServerMessage, callback: ((Data?) -> Void)? = nil) {
+        ipcQueue.async {
+            let result = try? self.ipcDriverClient?.requestRemote(message)
+            self.callbackQueue.async { callback?(result) }
```

Comment:
I believe normal URLRequest callbacks happen off of the main queue, so this is an attempt to mimic that in the hopes of catching more bugs with UI tests. It could very well increase flakiness of tests though.